keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -382,10 +388,107 @@ class ClusterHandler(
     return spec
   }
 
-  private fun List<ResourceDiff<ServerGroup>>.createsNewServerGroups() =
-    any {
-      !it.isCapacityOrAutoScalingOnly()
+  override suspend fun actuationInProgress(resource: Resource<ClusterSpec>) =
+    resource
+      .spec
+      .locations
+      .regions
+      .map { it.name }
+      .any { region ->
+        orcaService
+          .getCorrelatedExecutions("${resource.id}:$region")
+          .isNotEmpty()
+      }
+
+  /**
+   * Retrieve server group entity tags from clouddriver to identity the "source" of the server group (pipeline or
+   * orchestration), then retrieve execution details from orca to determine the deployment strategy used for that
+   * server group.
+   */
+  private suspend fun ServerGroup.discoverDeploymentStrategy(): ClusterDeployStrategy? {
+    return coroutineScope {
+      val entityTags = async {
+        log.debug("Looking for entity tags on server group $name in application ${moniker.app}, " +
+          "account ${location.account} in search of pipeline/task correlation.")
+        cloudDriverService.getEntityTags(
+          cloudProvider = "aws",
+          account = location.account,
+          application = moniker.app,
+          entityType = "servergroup",
+          entityId = name
+        )
+      }.await()
+
+      if (entityTags.isEmpty()) {
+        log.warn("Unable to find entity tags for server group $name in application ${moniker.app}, " +
+          "account ${location.account}.")
+        return@coroutineScope null
```

Comment:
What's the difference in behavior between `return` and `return@coroutineScope`?

Reply:
I had `return` originally but after adding `coroutineScope` up top I could no longer use that and had to specify the return point. I could rewrite this entire block as `return coroutineScope { ... }` without any returns inside, but the function is relatively long and I felt it would be less readable.